### PR TITLE
Add ValueError to jax.numpy.squeeze function.

### PR DESCRIPTION
**Pull request by us**
Like numpy.squeeze function, if the value given to the axis is not equal to 1, it raises a ValueError.
Related issue: #2284

### CODE REVIEW

File: jax/numpy/lax_numpy.py [hunk 1]
```diff
@@ -968,13 +968,21 @@ def ravel(a, order="C"):
...
     return a
   if axis is None:
     newshape = [d for d in shape(a) if d != 1]
   else:
     if isinstance(axis, int):
+      if shape(a)[axis] != 1:
+        raise ValueError(msg)
       axis = (axis,)
+    elif isinstance(axis, tuple):
+      for x in axis:
```
Comment (hawkinsp) [inline jax/numpy/lax_numpy.py]:
```
if any(shape(a)[x] != 1 for x in axis):
  raise ValueError(msg)
```

We like to minimize vertical space in JAX, and you can save a line here.

File: jax/numpy/lax_numpy.py [hunk 2]
```diff
@@ -968,13 +968,21 @@ def ravel(a, order="C"):
...
 def squeeze(a, axis=None):
+  msg = "cannot select an axis to squeeze out " \
+            "which has size not equal to one"
   if 1 not in shape(a):
     return a
   if axis is None:
     newshape = [d for d in shape(a) if d != 1]
   else:
     if isinstance(axis, int):
+      if shape(a)[axis] != 1:
```
Comment (hawkinsp) [inline jax/numpy/lax_numpy.py]: Couldn't you avoid duplicating this check by simply waiting until after the `axis = (axis,)` call below and then making the test unconditional?

This would also allow you to move `msg` closer to its use.

**Review by hawkinsp — CHANGES_REQUESTED** [review]
This looks fine overall, but there are a couple of minor improvements you could make.

File: jax/numpy/lax_numpy.py (hunk 2 above)
Comment (us) [reply jax/numpy/lax_numpy.py]: How can I check int and tuple same time?

Comment (hawkinsp) [reply jax/numpy/lax_numpy.py]: I mean you should write something like this:

```
axis = (axis,) if isinstance(axis, int) else axis
if any(shape(a)[x] != 1 for x in axis):
  raise ValueError("...")
```

**Review by hawkinsp — CHANGES_REQUESTED** [review]

File: jax/numpy/lax_numpy.py (hunk 2 above)
Comment (us) [reply jax/numpy/lax_numpy.py]: Okey! We are already converting to tuple if integer so two different queries are meaningless. I'm sorry to miss this. Thank you!

File: jax/numpy/lax_numpy.py [hunk 3]
```diff
@@ -968,13 +968,16 @@ def ravel(a, order="C"):
 
 @_wraps(onp.squeeze)
 def squeeze(a, axis=None):
+  msg = "cannot select an axis to squeeze out " \
+            "which has size not equal to one"
   if 1 not in shape(a):
```
Comment (GeorgOstrovski) [inline jax/numpy/lax_numpy.py]: this solution is a little incomplete: because of the "if 1 not in shape(a):" block, it will miss wrong axis specifications when there's no singleton dimension at all

**Comment by hawkinsp** [comment]
Thanks for the PR! I think #2470 fixed this case; closing.

Please reopen if I've misunderstood!
